packages/guides/src/Compiler/DocumentNodeTraverser.php

```diff
@@ -50,7 +50,7 @@ private function traverseForTransformer(
         }
 
         foreach ($shadowNode->getChildren() as $shadowChild) {
-            $this->traverseForTransformer($transformer, $shadowChild, $compilerContext);
+            $this->traverseForTransformer($transformer, $shadowChild, $compilerContext->withShadowTree($shadowChild));
```

Comment:
This was a bug, we should pass the current node in the shadow tree. So we can move around at the correct level.